src/CodeTracking.jl

```diff
@@ -186,17 +188,24 @@ see [`definition(Expr, method::Method)`](@ref) instead.
 """
 function definition(::Type{String}, method::Method)
     file, line = whereis(method)
-    src = read(file, String)
+    src = if startswith(file, "REPL[")
+        hist_idx = parse(Int, file[6:end-1])
+        hp = Base.active_repl.interface.modes[1].hist
+        src = hp.history[hp.start_idx+hist_idx]
+    else
+        read(file, String)
+    end
```

Comment:
I wonder if everything above this point should be put into it own function and exported? MagneticReadHead.jl right now has a function that looks exactly like that

And Debugger.jl almost does.

Reply:
I don't favor exporting it, but we can have this be the official implementation for everyone.